modern/src/map/Map.js

```diff
@@ -35,7 +35,27 @@ const updateReadyValue = value => {
   readyListeners.forEach(listener => listener(value));
 };
 
+const onIdle = function () {
+  const zoom = Math.round(map.getZoom())
+  if (zoom !== map.getZoom()) {
+    map.zoomTo(zoom)
+  }
+};
+
+const isRasterMap = map => Object.values(map.getStyle().sources).some(e => e.type === 'raster');
+
+const setAllowedZoomLevels = map => {
+  if(isRasterMap(map)) {
+    map.scrollZoom.setWheelZoomRate(1);
+    map.on('idle', onIdle);
+  } else {
+    map.scrollZoom.setWheelZoomRate(1/450);
+    map.off('idle', onIdle);
+  }
+};
+
 const initMap = async () => {
+  setAllowedZoomLevels(map);
```

Comment:
How do we remove the listener?

Reply:
line 53 `map.off('idle', onIdle);` removes the listener.

Reply:
When the map is first loaded or a map style change is made using the switcher, `setAllowedZoomLevels` gets called. It checks whether the map is using raster tiles, if it is, it adds an event listener. If it is not, it removes the event listener. Adding the same function for the same event listener multiple times does not result in duplicate event listeners. Or trying to remove an event listener that is not binded do not cause any error. I didn't understand why we need to remove the event listener when map is not displayed because the map itself handles firing of those listener functions, when the map is gone so are all the event listener functions binded to it.

Reply:
Map is never gone. It's only hidden. So, the right thing to do is to remove it. That's why we use react components for anything that's attached to the map.

Reply:
Why? We want the event listener function to be there when we want to see the map again. What kind of a problem can it cause?

Reply:
I can't think of any specific problem, but I also don't see any good reason to not do it the right way.

Reply:
OK. What about this: in `Map` component I listen for changes on `mapReady` state using `useEffect` hook. If the map is ready and uses raster tiles add the event listener. Return a function that will remove the event listener as a cleanup function. Sounds good?

Reply:
Not sure I understand. Can you please update PR?

Reply:
done.